src/main/java/it/pagopa/pagopa/apiconfig/service/CdiService.java

```diff
@@ -152,6 +141,224 @@ public void deleteCdi(String idCdi, String pspCode) {
+                        .value(channelId + " - " + informativaDetail.getTipoVersamento())
+                        .valid(CheckItem.Validity.NOT_VALID)
+                        .action("Canale non censito")
+                        .build());
+            }
+            else if (psp != null) {
+                // check payment methods
+                checkItemList.add(checkPaymentMethods(channel.get(), psp, informativaDetail));
+
+                if (informativaDetail.getModelloPagamento() == 4L) {
+                    // check payment model
+                    checkItemList.add(checkPaymentModel(informativaDetail));
+                }
+
+                // check broker psp
+                checkItemList.add(checkBrokerPsp(channel.get(), informativaDetail));
+            }
+
+            // check amount ranges
+            checkItemList.addAll(checkAmountRanges(informativaDetail));
+
+
+            // check languages
+            checkItemList.addAll(checkLanguages(informativaDetail));
+        }
+
+        return checkItemList;
+    }
+
+    private CheckItem checkPaymentMethods(Canali channel, Psp psp, CdiXml.InformativaDetail informativaDetail) {
+        List<PspCanaleTipoVersamento> paymentMethods = pspCanaleTipoVersamentoRepository.findByFkPspAndCanaleTipoVersamento_FkCanale(psp.getObjId(), channel.getId());
+        String paymentType = informativaDetail.getTipoVersamento() != null ? informativaDetail.getTipoVersamento() : "PO";
+
+        CheckItem.Validity validity = paymentMethods.stream().anyMatch(pm -> pm.getCanaleTipoVersamento().getTipoVersamento().getTipoVersamento().equals(paymentType)) ? CheckItem.Validity.VALID : CheckItem.Validity.NOT_VALID;
+        return CheckItem.builder()
+                .title("Canale - Tipo Versamento")
+                .value(channel.getIdCanale() + " - " + informativaDetail.getTipoVersamento())
+                .valid(validity)
+                .action(validity.equals(CheckItem.Validity.VALID) ? "" : "Canale e/o tipo versamento non coerenti con quanto censito")
+                .build();
+    }
+
+    private List<CheckItem> checkLanguages(CdiXml.InformativaDetail informativaDetail) {
+        List<String> languages = new ArrayList<>();
+        List<CheckItem> checkItemList = new ArrayList<>();
+        String title = "Codice lingua";
+
+        for(CdiXml.InformazioniServizio informazioniServizio: informativaDetail.getListaInformazioniServizio().getInformazioniServizio()) {
+            languages.add(informazioniServizio.getCodiceLingua());
+        }
+
+        if (languages.size() == 5) {
+            checkItemList.add(CheckItem.builder()
+                    .title(title)
+                    .value("")
+                    .valid(CheckItem.Validity.VALID)
+                    .build());
+        }
+        else if (languages.size() < 5) {
+            List<String> languagesTarget = Stream.of("IT", "EN", "DE", "FR", "SL").collect(Collectors.toList());
```

Comment:
I think this is better
```suggestion
            List<String> languagesTarget = List.of("IT", "EN", "DE", "FR", "SL");
```

Reply:
I can't use it because `List.of` returns an unmodifiable list.